packages/cli/src/commands/init/utils/repo.ts

```diff
@@ -96,25 +141,43 @@ export const configureGitSetup = (
           task.skip('Checkout out the branch failed.');
           throw new Error(err);
         })
+    },
+    {
+      title: 'Remove .git',
+      task: () => removeGit(dir)
+    },
+    {
+      title: 'Setup new git respository',
+      task: () => addGit(dir)
     }
   ];
-    // setup the main tasks launcher
+
+  // setup the main tasks launcher
   const tasks = new Listr(gitSetup, { concurrent: false });
   return tasks;
 };
 
+
+
+
 export const cloneRepo = async (designSystemOptions: any): Promise<boolean> => {
   try {
     const repoRef = selectRepo(designSystemOptions);
     const dir = path.join(process.cwd(), designSystemOptions['ds-name']);
     const definitiveRef: BranchRef | boolean = repoRef;
 
+    if(typeof definitiveRef === 'boolean'){
+      return false;
+    }
+
```

Comment:
Why not?
 ```
if(definitiveRef === false) {
  return false;
}
```

Reply:
If it's not a BranchRef we don't want to continue with this function.

Reply:
`repoRef` returns a boolean or`BranchRef`?

Reply:
Yea, selectRepo will return a git repo + branch, but if there is no repo, or it's coming soon repo it returns false.

Reply:
Hmm.. maybe within `repo.selectRepo()`, we can keep the return type similar to the union type declared for `repoRef: BranchRef | null`. 

The current condition that checks whether a repoRef is null, returns `false`. which is why `repo.selectRepo => BranchRef | boolean` 

```
  if (!repoRef) {
    logger.info('coming soon... 😉');
    return false;
  }
```

Might make it a little easier to know the expected return type from `repo.selectRepo`, and checking whether it's null or not. Typescript is too fancy lol  

```
if(definitiveRef === false) {
  return false;
}
```
to 

```
if(!definitiveRef) {
 return false;
}
```

Reply:
Thanks I changed it to this - keep the types consistant is a great idea.